### ISSUE TITLE
Skipping key stretching?

### ISSUE THREAD

**Issue by shesek** [keybase/triplesec#32]
I'm using keys that are randomly generated and not based on a user provided password. For this case, stretching the key doesn't add any value.

Is there a way to tell TripleSec to skip that?

**Comment by SparkDustJoe** [comment]
I don't think this made it into V4 #51 and come to think of it, in my C# port I don't think I put it in either... Not sure if this is still needed, but I'll bump this up to the discussion for V5 #72 . 
The two schools of thought proposed here (skimming the comments) are:

1) To have functions that allow you to skip the keying altogether, and have methods that require the full key bytes to be passed in (which puts the onus on the implementer to make secure key material), or...
2) To have a "reduced" KDF (be it Scrypt or Argon2 or whatever) which will still produce the internal key bytes needed, but would allow faster server or transport keys and reduce the overhead for automatically handled files rather than "human accessed" files. This could also allow a master key for a GROUP of files, each generating their own salt and internal keys based on one master key, and each file then doesn't need to take 2-5 seconds each on pre-processing.  The master key has to be strong enough to hold up to brute-force attacks (using the full KDF compute since we're likely dealing with human input), and the output has to be long enough that ONLY exhaustive search or a collision would produce the key that unlock the individual files (the theory being since the files use a "cheap" KDF vs. the master, it would be "easier" to attack that, but the file key would still be very long going through that "cheap" KDF so the attacker isn't really catching a break; this strength then puts the onus on the individual encryption keys, which should be strong in the first place otherwise we're chasing our tails).
